CesiumGeospatial/test/TestSimplePlanarEllipsoidCurve.cpp

```diff
@@ -0,0 +1,87 @@
+#include <CesiumGeospatial/SimplePlanarEllipsoidCurve.h>
+
+#include <catch2/catch.hpp>
+
+using namespace CesiumGeospatial;
+using namespace CesiumUtility;
+
+TEST_CASE("SimplePlanarEllipsoidCurve") {
```

Comment:
These tests look like a great start.

A suggestion for the future...

Currently it looks like each test section tests our assumptions of what a flight should be
  1) Is the start and end ok?
  2) Is it interpolating sensibly?
  3) Etc

We may want to flip this around so that each section tests a specific flight of interest, with each flight tested against our assumptions.
   1) Philly to Tokyo
   2) Tokyo To Philly
   3) Philly to Philly (same start / dest)
   4) Flight created from SimplePlanarEllipsoidCurve::fromEarthCenteredEarthFixedCoordinates
   5) Flight created from SimplePlanarEllipsoidCurve::fromLongitudeLatitudeHeight
   6) Flight from center of the earth
   7) etc.

Reply:
@csciguy8 I've expanded the tests based on your list here. One note is that a flight from the center of the earth shouldn't work - it's the one position that can't be mapped to a geocentric surface, as it'll result in +Infinity when calculating `1.0 / sqrt(0.0)`. CesiumJS also returns NaN for this input to `scaleToGeocentricSurface`, so I've made this a condition to fail the construction of a `SimplePlanarEllipsoidCurve` and added a test for it.